metricpublisher/schemas/metric_datum.json

```diff
@@ -0,0 +1,98 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Schema for the metric_datum request object",
+    "type": "object",
+    "properties": {
+        "MetricName": {
+            "type": "string"
+        },
+        "Dimensions": {
+            "type": "array",
+            "properties": {
+                "Name": {
+                    "type": "string"
+                },
+                "Value": {
+                    "type": "string"
+                }
+            },
+            "required": [
+                "Name",
+                "Value"
+            ]
+        },
+        "Timestamp": {
+            "type": "number"
+        },
+        "Value": {
+            "type": "number"
+        },
+        "StatisticValues": {
+            "type": "object",
+            "properties": {
+                "SampleCount": {
+                    "type": "number"
+                },
+                "Sum": {
+                    "type": "number"
+                },
+                "Minimum": {
+                    "type": "number"
+                },
+                "Maximum": {
+                    "type": "number"
+                }
+            },
+            "required": [
+                "SampleCount",
+                "Sum",
+                "Minimum",
+                "Maximum"
+            ]
+        },
+        "Unit": {
+            "type": "string",
+            "enum": [
+                "Seconds",
+                "Microseconds",
+                "Milliseconds",
+                "Bytes",
+                "Kilobytes",
+                "Megabytes",
+                "Gigabytes",
+                "Terabytes",
+                "Bits",
+                "Kilobits",
+                "Megabits",
+                "Gigabits",
+                "Terabytes",
+                "Bits",
+                "Kilobits",
+                "Megabits",
+                "Gigabits",
+                "Terabits",
+                "Percent",
+                "Count",
+                "Bytes/Second",
+                "Kilobytes/Second",
+                "Megabytes/Second",
+                "Gigabytes/Second",
+                "Terabytes/Second",
+                "Bits/Second",
+                "Kilobits/Second",
+                "Megabits/Second",
+                "Gigabits/Second",
+                "Terabits/Second",
+                "Count/Second",
+                "None"
+            ]
+        },
+        "StorageResolution": {
+            "type": "number"
+        }
+    },
+    "required": [
```

Comment:
Did you try these two cases with cli: 1. both Value and StatisticValues exist 2. both Value and StatisticValues don't exist?

Reply:
Yes - both of them give an error. I was unaware that they were supposed to be collectively exhaustive and mutually exclusive. Should this be reflected in the JSON schema?

Reply:
Yes.